### ISSUE TITLE
Upload images/videos from any iOS app

### ISSUE THREAD

**Issue by EddyLB** [Piwigo/Piwigo-Mobile#284]
It would be useful to be able to upload images to a Piwigo server from any iOS app.
Such feature requires the development of an app extension accessing common settings stored in a shared container, typically the URL, the username and default upload settings.

**Comment by paraviz02** [comment]
@EddyLB - is this functionality the same as being able to upload from the Apple Photos app on iOS to Facebook, or to a text message, etc?

**Comment by EddyLB** [comment]
Hi @paraviz02 
If I understand well, the functionality you mention is already available. With the Piwigo Mobile app, you can already share photos and videos stored on your Piwigo server, including sending by email or with a text message, printing and saving photos of your Piwigo server into you camera roll.
Here, the idea is to add the ability to upload photos/videos to your Piwigo server from any iOS app other than the Piwigo iOS app.

**Comment by paraviz02** [comment]
@EddyLB - sorry I totally missed this ... How about being able to select/send photos from Apple Photos to a Piwigo album?

For example, take a photo with the Camera app, tap on the image afterwards, hit the share button and select Piwigo. Then select the Piwigo album to upload to and send it off. (With possibly the ability to edit the title etc.)

**Comment by EddyLB** [comment]
Yes, that would be one of the many possibilities. We also envisage to provide the capability to take photos directly from the Piwigo iOS app, with photos being uploaded automatically to the current album.

**Comment by paraviz02** [comment]
Ah sweet. Both of those items would be very productive! We just got back from a get-together with extended family, and it is upload time ... :-) Piwigo and iOS app are wonderful. Thanks for all your efforts!

**Comment by EddyLB** [comment]
To be implemented when albums will be stored in a persistent database.

**Comment by snohko** [comment]
Exactly. 
That’s the reason why I set up Piwigo in the first place. 
It’s ridiculous that Piwigo doesn’t appear in the share menu when you take a picture.

👎👎

awful 

// hko

_________
> For example, take a photo with the Camera app, tap on the image afterwards, hit the share button and select Piwigo. Then select the Piwigo album to upload to and send it off. (With possibly the ability to edit the title etc.)

**Comment by EddyLB** [comment]
Now that background uploads are available thanks to a new class introduced by Apple with iOS 26 (v4.2), we start working on this long-awaited feature…
